### PR TITLE
Add entries to `Designation` event

### PR DESCRIPTION
**Pull request by shargon**
# Description

Wait for more methods to create a new fork
Require https://github.com/neo-project/neo/pull/3362
Close https://github.com/neo-project/neo/issues/3389

## Type of change

- [ ] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Pending

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/SmartContract/Native/RoleManagement.cs [hunk 1]
```diff
@@ -26,7 +26,16 @@ public sealed class RoleManagement : NativeContract
     {
         [ContractEvent(0, name: "Designation",
             "Role", ContractParameterType.Integer,
-            "BlockIndex", ContractParameterType.Integer)]
+            "BlockIndex", ContractParameterType.Integer,
+            Hardfork.HF_Domovoi)]
```
Comment (AnnaShaleva) [inline src/Neo/SmartContract/Native/RoleManagement.cs:30]: The next one supposed to be HF_Echidna, may be borrowed from https://github.com/neo-project/neo/pull/3175/commits/4d5cee903fa77965d590a131243136b227c75c4b.

**Review by AnnaShaleva — COMMENTED** [review]
LGTM, it would be nice to have a unit-test.

**Comment by shargon** [comment]
The ut added also test that https://github.com/neo-project/neo/pull/3362 works fine :)

**Comment by vncoelho** [comment]
I think this is worth proceeding.

**Comment by shargon** [comment]
> @shargon how about this pr? will need the hardfork somewhere else.

Is waiting for more features in the same hardfork

**Comment by Jim8y** [comment]
BTW, please resolve conflicts.

**Comment by superboyiii** [comment]
Test: PASS
Before HF_Echidna:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "txid": "0xd7039f3f8317b461097ca1ff393f3f39b5ea05276c790603b703b44e40784421",
        "executions": [
            {
                "trigger": "Application",
                "vmstate": "HALT",
                "exception": null,
                "gasconsumed": "2090550",
                "stack": [
                    {
                        "type": "Any"
                    }
                ],
                "notifications": [
                    {
                        "contract": "0x49cf4e5378ffcd4dec034fd98a174c5491e395e2",
                        "eventname": "Designation",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "4"
                                },
                                {
                                    "type": "Integer",
                                    "value": "42"
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

After HF_Echidna:
```json
{
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
        "txid": "0x08a54e3183a2dd0f1e88303080b0b52d105a92c640c5504240d8863f1b14ea6b",
        "executions": [
            {
                "trigger": "Application",
                "vmstate": "HALT",
                "exception": null,
                "gasconsumed": "2090550",
                "stack": [
                    {
                        "type": "Any"
                    }
                ],
                "notifications": [
                    {
                        "contract": "0x49cf4e5378ffcd4dec034fd98a174c5491e395e2",
                        "eventname": "Designation",
                        "state": {
                            "type": "Array",
                            "value": [
                                {
                                    "type": "Integer",
                                    "value": "4"
                                },
                                {
                                    "type": "Integer",
                                    "value": "61"
                                },
                                {
                                    "type": "Array",
                                    "value": [
                                        {
                                            "type": "ByteString",
                                            "value": "AhJGxC9c3mJD/hg7FDYdohTlmdGyXPQoIA9kYK6qdBvh"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "AyHoH24FT3hSY3V9CZUzX1bSfJoXCfvds2sNNxVK6Boj"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "A111yY9zJv/n6nJrdzJbl7wG2NAIqzTZV1DJ22jYIMSP"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "A3briYQGk9je1GivNmTbNp0Av+2iHO7vFhX1Z66AzCMu"
                                        }
                                    ]
                                },
                                {
                                    "type": "Array",
                                    "value": [
                                        {
                                            "type": "ByteString",
                                            "value": "AxLpQlEVqhOWOqj0n3lsbSsjCHmuqD4XRYOHidQpCvwQ"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "AtbQ1aBGz24M4yvvNg+DNdAulA9//3VwKb2MTSyHgAGD"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "ArNnumlmTbihVD7I/jiNjx1Gl9SU/TZA+lRivjqm7Y9K"
                                        },
                                        {
                                            "type": "ByteString",
                                            "value": "AlT/v0cbtmoZSJu6jfuPyMYo/lHralETJfkny2c5REB7"
                                        }
                                    ]
                                }
                            ]
                        }
                    }
                ]
            }
        ]
    }
}
```

**Review by superboyiii — APPROVED** [review]

**Comment by shargon** [comment]
Merged into `HF_Echidna`

**Pull request merged**